src/client/component/workshop.cpp

```diff
@@ -72,6 +63,113 @@ namespace workshop
 		}
 	}
 
+	std::string get_mod_name(const std::string& mod_id)
+	{
+		if (mod_id == "usermaps")
+		{
+			return mod_id;
+		}
+
+		const utils::nt::library host{};
+		std::string path;
+
+		if (game::is_server())
+		{
+			const auto base_path = host.get_folder().generic_string();
+			path = utils::string::va("%s/mods/%s/zone/workshop.json", base_path.data(), mod_id.data());
+		}
+		else
+		{
+			return mod_id;
+		}
+
+		std::string json_str;
+		utils::io::read_file(path, &json_str);
+
+		if (json_str.empty())
+		{
+			printf("[ Workshop ] workshop.json has not been found in mod folder: %s\n", mod_id.data());
+			return mod_id;
+		}
+
+		rapidjson::Document doc;
+		const rapidjson::ParseResult parse_result = doc.Parse(json_str);
+
+		if (parse_result.IsError() || !doc.IsObject())
+		{
+			printf("[ Workshop ] Unable to parse workshop.json\n");
+			return mod_id;
+		}
+
+		if (doc.HasMember("Title"))
+		{
+			std::string title = doc["Title"].GetString();
+
+			if (title.size() > 31)
+			{
+				title.resize(31);
+			}
+
+			return title;
+		}
+		else
+		{
+			printf("[ Workshop ] workshop.json has no \"Title\" member.\n");
+			return mod_id;
+		}
```

Comment:
you can remove the else scope here, as the if has a return statement
```suggestion

		printf("[ Workshop ] workshop.json has no \"Title\" member.\n");
		return mod_id;
```